src/main/kotlin/uk/gov/dluhc/printapi/rds/entity/Certificate.kt

```diff
@@ -94,28 +93,39 @@ class Certificate(
 
     fun addPrintRequest(newPrintRequest: PrintRequest): Certificate {
         printRequests += newPrintRequest
+        assignStatus()
         return this
     }
 
-    @PrePersist
```

Comment:
Refactored as the `@PreUpdate` was not invoked reliably as the changes to the associated print request and print request statuses does not cause any changes in the Certificate for the hook to be called that set the correct status.  Status is now managed in directly when print requests are added to the certificate or a new status is added to the current print request through the Certificate addStatus method.